src/deploy/firestore/release.ts

```diff
@@ -1,25 +1,33 @@
-import * as _ from "lodash";
-
 import { RulesDeploy, RulesetServiceType } from "../../rulesDeploy";
+import { logger } from "../../logger";
 import { Options } from "../../options";
-import { FirebaseError } from "../../error";
+import { RulesContext } from "./prepare";
 
 /**
  * Releases Firestore rules.
  * @param context The deploy context.
  * @param options The CLI options object.
  */
 export default async function (context: any, options: Options): Promise<void> {
-  const rulesDeploy: RulesDeploy = _.get(context, "firestore.rulesDeploy");
+  const rulesDeploy: RulesDeploy = context?.firestore?.rulesDeploy;
   if (!context.firestoreRules || !rulesDeploy) {
     return;
   }
 
-  const rulesFile = options.config.src.firestore?.rules;
-  if (!rulesFile) {
-    throw new FirebaseError(
-      `Invalid firestore config: ${JSON.stringify(options.config.src.firestore)}`
-    );
-  }
-  await rulesDeploy.release(rulesFile, RulesetServiceType.CLOUD_FIRESTORE);
+  const rulesContext: RulesContext[] = context?.firestore?.rules;
+  await Promise.all(
+    rulesContext.map(async (ruleContext: RulesContext): Promise<void> => {
+      const databaseId = ruleContext.databaseId;
+      const rulesFile = ruleContext.rulesFile;
+      if (!rulesFile) {
+        logger.error(
+          `Invalid firestore config for ${databaseId} database: ${JSON.stringify(
+            options.config.src.firestore
+          )}`
+        );
+        return;
+      }
+      await rulesDeploy.release(rulesFile, RulesetServiceType.CLOUD_FIRESTORE, databaseId);
```

Comment:
Since ruleDeploy.release returns Promise<void>, i think you can return it instead of awaiting it here
```suggestion
      return rulesDeploy.release(rulesFile, RulesetServiceType.CLOUD_FIRESTORE, databaseId);
```